src/bpfmap.cpp

```diff
@@ -60,17 +54,143 @@ bool BpfMap::is_per_cpu_type() const
          type() == libbpf::BPF_MAP_TYPE_LRU_PERCPU_HASH;
 }
 
-bool BpfMap::is_clearable() const
-{
-  return is_bpf_map_clearable(type());
-}
-
 bool BpfMap::is_printable() const
 {
   // Internal maps are not printable
   return bpf_name().starts_with("AT_");
 }
 
+KeyVec BpfMap::collect_keys() const
+{
+  uint8_t *old_key = nullptr;
+  auto key = KeyType(key_size_);
+
+  // snapshot keys, then operate on them
+  KeyVec keys;
+  while (bpf_map_get_next_key(fd(), old_key, key.data()) == 0) {
+    keys.push_back(key);
+    old_key = key.data();
+  }
+  return keys;
+}
+
+Result<OK> BpfMap::zero_out(int nvalues) const
```

Comment:
nit
```suggestion
Result<> BpfMap::zero_out(int nvalues) const
```
As was just pointed out to me, the OK is the default so you don't need to type it out explicitly.

Reply:
Same for the ones below.

Reply:
Done